core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -105,31 +105,31 @@ void findNewAndAddDanglingIndices(MetaData metaData) {
      * as dangled already.
      */
     Map<String, IndexMetaData> findNewDanglingIndices(MetaData metaData) {
-        final Set<String> indices;
+        final Set<String> indicesUUIDs;
         try {
-            indices = nodeEnv.findAllIndices();
+            indicesUUIDs = nodeEnv.findAllIndices();
```

Comment:
I think we need to fix this too that we don't read the index name from the directory name. We have to read some descriptor which I think is available everywhere now if not it's not an index. Then when we have that we can just use Index.java as a key